### PR TITLE
centrifuge-gsrpc-v2 milestone 2 evaluation

### PR DESCRIPTION
**Pull request by niklabh**
# Milestone Delivery Checklist

- [x] The [milestone-delivery-template.md](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/deliveries/milestone-delivery-template.md) has been copied and updated.
- [ ] The [invoice form :pencil:](https://forms.gle/LSRr7PCjBpEbKGh89) has been filled out for this milestone.
- [ ] This pull request is being made by the same account as the accepted application.
- [ ] I have disclosed any and all sources of reused code in the submitted repositories and have done my due diligence to meet its license requirements.
- [ ] In case of acceptance, the payment will be transferred to the BTC/ETH/fiat account provided in the application.
- [ ] The delivery is according to the [Guidelines for Milestone Deliverables](https://grants.web3.foundation/docs/Support%20Docs/milestone-deliverables-guidelines).

Link to the application pull request: https://github.com/w3f/Grant-Milestone-Delivery/pull/890

### CODE REVIEW

File: evaluations/centrifuge-gsrpc-v2_2_niklabh.md [hunk 1]
```diff
@@ -0,0 +1,574 @@
...
+| 6.     | In-Memory Registry               | <ul><li>[x] </li> </ul> | [centrifuge/go-substrate-rpc-client#338](https://github.com/centrifuge/go-substrate-rpc-client/pull/338)                                                                          | In-Memory Registry is implemented                                                                                         |
+
+## General Notes v1 - June 26, 2023
+
+- Documentation needs to be improved. Basic tutorial on how to parse any substrate event and type using the new version of the library is missing.
+- Unit test coverage is good.
+- There is an error in running tests in Makefile. I have submitted a [PR](https://github.com/centrifuge/go-substrate-rpc-client/pull/353) to fix the same.
+- Live tests are running for too long. No need to check for 1000 events. 50 are fine.
```
Comment (mikiquantum) [inline evaluations/centrifuge-gsrpc-v2_2_niklabh.md:26]: This test is very useful not only for the milestone, but to run as a health check of the API over time. What we will do is to default to 50 but pass an env var to override this value at will.

File: evaluations/centrifuge-gsrpc-v2_2_niklabh.md [hunk 2]
```diff
@@ -0,0 +1,574 @@
...
+| 5.     | Dockerfile                       | <ul><li>[x] </li> </ul> | [Dockerfile](https://github.com/centrifuge/go-substrate-rpc-client/blob/b9da36ebf177cbb3ee41c2ef2b9a1fb08e5f8522/Dockerfile_milestone2)                                           | Dockerfile is provided                                                                                                    |
+| 6.     | In-Memory Registry               | <ul><li>[x] </li> </ul> | [centrifuge/go-substrate-rpc-client#338](https://github.com/centrifuge/go-substrate-rpc-client/pull/338)                                                                          | In-Memory Registry is implemented                                                                                         |
+
+## General Notes v1 - June 26, 2023
+
+- Documentation needs to be improved. Basic tutorial on how to parse any substrate event and type using the new version of the library is missing.
+- Unit test coverage is good.
+- There is an error in running tests in Makefile. I have submitted a [PR](https://github.com/centrifuge/go-substrate-rpc-client/pull/353) to fix the same.
```
Comment (mikiquantum) [inline evaluations/centrifuge-gsrpc-v2_2_niklabh.md:25]: Thank you

File: evaluations/centrifuge-gsrpc-v2_2_niklabh.md [hunk 3]
```diff
@@ -0,0 +1,574 @@
...
+| 5.     | Dockerfile                       | <ul><li>[x] </li> </ul> | [Dockerfile](https://github.com/centrifuge/go-substrate-rpc-client/blob/b9da36ebf177cbb3ee41c2ef2b9a1fb08e5f8522/Dockerfile_milestone2)                                           | Dockerfile is provided                                                                                                    |
+| 6.     | In-Memory Registry               | <ul><li>[x] </li> </ul> | [centrifuge/go-substrate-rpc-client#338](https://github.com/centrifuge/go-substrate-rpc-client/pull/338)                                                                          | In-Memory Registry is implemented                                                                                         |
+
+## General Notes v1 - June 26, 2023
+
+- Documentation needs to be improved. Basic tutorial on how to parse any substrate event and type using the new version of the library is missing.
```
Comment (mikiquantum) [inline evaluations/centrifuge-gsrpc-v2_2_niklabh.md:23]: The `retriever.GetEvents(blockHash)` is the one in charge of parsing the events of a block into the `parser.Event` struct. How to use it is described [here](https://github.com/centrifuge/go-substrate-rpc-client/blob/20047b8b15d046d0fa137cc716cb489395a0081a/registry/retriever/event_retriever_live_test.go#L70) as stated in the REGISTRY.md. 
Are you referring to the deserialization/unmarshalling of those events into a user defined (strongly typed) structs?

**Review by mikiquantum — COMMENTED** [review]
@niklabh Thanks for the review.

**Comment by semuelle** [comment]
Thanks for the evaluation, @niklabh. Nice work. I just left a clarifying comment on one of the issues. Anyway, I'll merge the evaluation as it is. Feel free to submit an update when the issues are resolved. Otherwise, we'll take it from here.

File: evaluations/centrifuge-gsrpc-v2_2_niklabh.md (hunk 3 above)
Comment (mikiquantum) [reply evaluations/centrifuge-gsrpc-v2_2_niklabh.md:23]: OK! Thanks @semuelle. Added the changes requested in this [commit](https://github.com/centrifuge/go-substrate-rpc-client/commit/7e3e5ad5225bfa7afd342f3fe8932dc44702c472) and updated the delivery.

File: evaluations/centrifuge-gsrpc-v2_2_niklabh.md (hunk 1 above)
Comment (niklabh) [reply evaluations/centrifuge-gsrpc-v2_2_niklabh.md:26]: Thanks

**Comment by niklabh** [comment]
Updated evaluation: https://github.com/w3f/Grant-Milestone-Delivery/pull/897/files

**Comment by RouvenP** [comment]
hi @niklabh we just transferred the KSM

**Pull request merged**